tests/run/i21346.scala

```diff
@@ -0,0 +1,10 @@
+// scalajs: --skip
+
+object Foo:
+    def foo[X, Y, Z](x: X, y: Y)[A](z: Z) = ???
```

Comment:
We need to test what happens if we actually refer to `A`. And perhaps also give a result type since `Nothing` is a bit weird and not what we're testing for:
```suggestion
    def foo[X, Y, Z](x: X, y: Y)[A](z: Z, A: A): (X, Y, Z, A) = ???
```

Reply:
Likewise, we should test what happens if the bounds of a type parameter are path-dependent on a previous term parameter.

It might not result in something very sensible, but it should be good enough not to crash Java reflection.